File: Server/src/main/java/com/api/server/foodb/FoodcomexCompoundProvider.java
package com.api.server.foodb;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "foodcomex_compound_providers")
public class FoodcomexCompoundProvider {
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @NotNull
    @Column(name = "foodcomex_compound_id", nullable = false)
    private Integer foodcomexCompoundId;

    @NotNull
    @Column(name = "provider_id", nullable = false)
    private Integer providerId;

    @Column(name = "created_at")
    private Instant createdAt;

    @Column(name = "updated_at")
    private Instant updatedAt;

}

File: DotNetServer/DotNetServer/Core/Entities/Foodb/FoodcomexCompoundProvider.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace DotNetServer.Core.Entities.Foodb;

[Table("foodcomex_compound_providers")]
[Index("FoodcomexCompoundId", Name = "index_foodcomex_compound_providers_on_foodcomex_compound_id")]
[Index("ProviderId", Name = "index_foodcomex_compound_providers_on_provider_id")]
public partial class FoodcomexCompoundProvider
{
    [Key] [Column("id")] public int Id { get; set; }

    [Column("foodcomex_compound_id")] public int FoodcomexCompoundId { get; set; }

    [Column("provider_id")] public int ProviderId { get; set; }

    [Column("created_at", TypeName = "datetime")]
    public DateTime? CreatedAt { get; set; }

    [Column("updated_at", TypeName = "datetime")]
    public DateTime? UpdatedAt { get; set; }
}